File: Lab1/ts_solution/src/modules/FileReader.ts
import { promises as fs } from "fs";
import { TextData } from "./TextData";
export class FileReader {
    static async ReadFileData(paths: string[]): Promise<TextData[]> {
        const dataList: TextData[] = [];

        for (const path of paths) {
            try {
                const text = await fs.readFile(path, "utf-8");
                dataList.push(TextData.ProcessText(path, text));
            } catch (err) {
                throw new Error(`Error occurred: ${err}`);
            }
        }
        return dataList;
    }
}

File: Lab1/cs_solution/FileReader.cs
class FileReader{
    public static List<TextData> ReadFileData(string[] paths){
        List<TextData> dataList = new List<TextData>();
        foreach(var path in paths){
            try{
                string text = File.ReadAllText(path);
                dataList.Add(TextData.ProcessText(Path.GetFileName(path), text));
            }catch(FileNotFoundException err){
                throw new FileNotFoundException($"Error: File {path} can't be found \n{err.Message}");
            }
        } 
        return dataList;
    }
}
